homeassistant/components/recorder/__init__.py

```diff
@@ -177,8 +179,12 @@ def run(self):
                           message="Error during connection setup: %s")
 
         if self.purge_days is not None:
-            track_point_in_utc_time(self.hass,
-                                    lambda now: self._purge_old_data(),
+            def purge_ticker(event):
+                """Rerun purge every second day."""
+                self._purge_old_data()
+                track_point_in_utc_time(self.hass, purge_ticker,
+                                        dt_util.utcnow() + timedelta(days=2))
+            track_point_in_utc_time(self.hass, purge_ticker,
```

Comment:
Maybe there is a better way to reschedule this maintenance task?

A problem I see here is that the recorder instance will never be released with a config reload (Although we don't do this yet, its probably coming one of these days)